akka-actor/src/main/scala/akka/io/dns/internal/DnsClient.scala

```diff
@@ -105,18 +116,25 @@ import scala.util.Try
       log.debug(s"Decoded: $msg")
       // TODO remove me when #25460 is implemented
       if (msg.flags.isTruncated) {
-        log.warning("DNS response truncated and fallback to TCP is not yet implemented. See #25460")
+        log.debug("DNS response truncated, falling back to TCP")
+        inflightRequests.get(msg.id) match {
+          case Some((_, msg)) ⇒
+            tcpDnsClient ! msg
+          case _ ⇒
+            log.debug("Client for id {} not found. Discarding unsuccessful response.", msg.id)
+        }
+      } else {
+        val (recs, additionalRecs) = if (msg.flags.responseCode == ResponseCode.SUCCESS) (msg.answerRecs, msg.additionalRecs) else (Nil, Nil)
+        self ! Answer(msg.id, recs, additionalRecs)
```

Comment:
Why do we need to reschedule it a message, couldn't we just deal with the result right away?

Reply:
in the truncated case, it's the `TcpDnsClient` that sends the `Answer` to the `DnsClient`

Reply:
Could it not get the replyTo out of the inflightRequests then send it back right away?